src/mappers/infrastructure/infrastructure_mapper.cpp

```diff
@@ -0,0 +1,84 @@
+#include "src/mappers/infrastructure/infrastructure_mapper.h"
+
+#include <cmath>
+#include <numeric>
+
+#include "src/vic3_world/states/vic3_state.h"
+#include "external/fmt/include/fmt/format.h"
+
+// Find root of a function fn using secant approximation.
+template <typename T, typename U>
+U find_root(const std::function<U(T)>& fn, T x_0, const std::function<T(U)>& x_1_func, int iterations)
+{
+   U y_0 = fn(x_0);
+   T x_1 = x_1_func(y_0);
+   U y_1 = fn(x_1);
+
+   for (; iterations > 0 && std::abs(y_1) > 0.005; --iterations)
+   {
+      float new_x = x_1 - (y_1 * (x_1 - x_0) / (y_1 - y_0));
+
+      x_0 = x_1;
+      y_0 = y_1;
+      x_1 = new_x;
+      y_1 = fn(new_x);
+   }
+   return x_1;
+}
+
+namespace mappers
+{
+InfrastructureMapper::InfrastructureMapper(const std::map<int, vic3::State>& states)
+{
+   target_hoi_infra_per_state_ = 1997.0/899;  // TODO: put this in a file somewhere
```

Comment:
perhaps this should go in a configurables/states.txt file? and have things like world manpower, infrastructure per state, etc. defined there so users can easily adjust if they want a high manpower/infrastructure/etc game, instead of having to adjust for each state individually